intro/intro_tutorial.py

```diff
@@ -0,0 +1,248 @@
+# For more detail on the process of developing labeling functions and other training data operators in Snorkel, see the [Introductory tutorial](#).
+
+# %% [markdown]
+# ## (2) Combining & Cleaning the Labels
+#
+# Our next step is to apply the labeling functions we wrote to the unlabeled training data.
+# The result is a _label matrix_, `L_train`, where each row corresponds to a data point and each column corresponds to a labeling function.
+# Since the labeling functions have unknown accuracies and correlations, their output labels may overlap and conflict; we use the `LabelModel` to automatically estimate their accuracies and correlations, reweight and combine their labels, and produce our final set of clean, integrated training labels:
+
+# %%
+from snorkel.labeling import PandasLFApplier, LabelModel
+
+# Define the set of labeling functions (LFs)
+lfs = [lf_keyword_my, lf_regex_check_out, lf_short_comment, lf_textblob_polarity]
+
+# Apply the LFs to the unlabeled training data
+applier = PandasLFApplier(lfs)
+L_train = applier.apply(df_train)
+
+# Train the label model and compute the training labels
+label_model = LabelModel(cardinality=2, verbose=True)
+label_model.fit(L_train, n_epochs=500, log_freq=50, seed=123)
+df_train["label"] = label_model.predict(L=L_train, tie_break_policy="abstain")
+
+# %% [markdown]
+# Note that we used the `LabelModel` to label data; however, on many data points, all the labeling functions abstain, and so the `LabelModel` abstains as well.
+# We'll filter these examples out of our training set now:
+
+# %%
+df_train = df_train[df_train.label != ABSTAIN]
+
+# %% [markdown]
+# Our ultimate goal is to use the resulting labeled training data points to train a machine learning model that can **generalize beyond the coverage of the labeling functions and the `LabelModel`**.
+# However first we'll explore some of Snorkel's other operators for building and managing training data.
+
+# %% [markdown]
+# ## (3): Writing Transformation Functions for Data Augmentation
+#
+# An increasingly popular and critical technique in modern machine learning is [data augmentation](#), the strategy of artificially _augmenting_ existing labeled training datasets by creating transformed copies of the data points.
+# Data augmentation is a practical and powerful method for injecting information about domain invariances into ML models via the data, rather than by trying to modify their internal architectures.
+# The canonical example is randomly rotating, stretching, and transforming images when training image classifiers---a ubiquitous technique in the field of computer vision today.
+# However, data augmentation is increasingly used in a range of settings, including text.
+#
+# Here, we implement a simple text data augmentation strategy- randomly replacing a word with a synonym.
+# We express this as a *transformation function (TF)*:
+
+# %%
+from snorkel.augmentation import transformation_function
```

Comment:
isort isn't configured right now (doesn't work on cell-by-cell imports in notebooks), but please do a quick manual pass